tests/bud.bats

```diff
@@ -2837,6 +2837,25 @@ EOF
   expect_output --substring "\[2/2] STEP 1/2: FROM busybox"
 }
 
+# Following test must fail since we are trying to run linux/arm64 on linux/amd64
+# Issue: https://github.com/containers/buildah/issues/3712
+@test "build-with-inline-platform" {
+  # Tries building image where baseImage has --platform=linux/arm64
+  run_buildah 1 build --signature-policy ${TESTSDIR}/policy.json -t test ${TESTSDIR}/bud/platform
+  expect_output --substring "Exec format error"
```

Comment:
This passes on a system with `qemu-user-static` installed. (As in: the build passes, so the test fails).

Reply:
Friendly suggestion:
```diff
diff --git a/tests/bud.bats b/tests/bud.bats
index 99066bb7..ba5cd4b9 100644
--- a/tests/bud.bats
+++ b/tests/bud.bats
@@ -2840,9 +2840,26 @@ EOF
 # Following test must fail since we are trying to run linux/arm64 on linux/amd64
 # Issue: https://github.com/containers/buildah/issues/3712
 @test "build-with-inline-platform" {
-  # Tries building image where baseImage has --platform=linux/arm64
-  run_buildah 1 build --signature-policy ${TESTSDIR}/policy.json -t test ${TESTSDIR}/bud/platform
-  expect_output --substring "Exec format error"
+  # Host arch
+  run_buildah info --format '{{.host.arch}}'
+  myarch="$output"
+  for try in amd64 arm64 ppc64le s390x; do
+    if [[ "$otherarch" != "$myarch" ]]; then
+      otherarch="$myarch"
+      break
+    fi
+  done
+  # ...create a Containerfile with --platform=linux/$otherarch
+  run_buildah '?' build --signature-policy ${TESTSDIR}/policy.json -t test ${TESTSDIR}/bud/platform
+  if [[ $status -eq 0 ]]; then
+    # Build succeeded: we must have qemu-user-static installed. Confirm that
+    # the 'uname' emits what we expect
+    expect_output --from="${lines[-1]}" "$otherarch"
+    # FIXME: also inspect the image and confirm the arch
+  else
+    # Build failed: we DO NOT have qemu-user-static installed.
+    expect_output --substring "Exec format error"
+  fi
 }

 # Following test must pass since we want to tag image as arm64
```
Writing the Containerfile and updating the comments and updating the next test are left as an exercise for the reader.